### PR TITLE
Page move dialog: disable invalid page options (Volume 2)

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Enhancements

- The page move dialog now disables all pages that are invalid new parents for the page 

## Features

- Each Fiber area can now define additional requests for simple data endpoints or actions.

### CODE REVIEW

File: config/areas/site/requests.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+use Kirby\Cms\Find;
+
+return [
+	'tree' => [
+		'pattern' => 'site/tree',
+		'action'  => function () {
+			$parent = Find::parent(get('parent', 'site'));
+			$move   = get('move') ? Find::parent(get('move')) : null;
```
Comment (distantnative) [inline config/areas/site/requests.php]:
```suggestion
		'action'  => function () {
			$request = App::instance()->request();
			$parent  = Find::parent($request->get('parent', 'site'));
			$move    = $request->get('move');
			$move    = $move ? Find::parent($move) : null;
```

File: config/areas/site/requests.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			$pages  = [];
+
+			foreach ($parent->childrenAndDrafts() as $child) {
+				$panel = $child->panel();
+
+				$pages[] = [
+					'children'    => $panel->url(true),
+					'disabled'    => $move !== null && $move->isMovableTo($child) === false,
```
Comment (distantnative) [inline config/areas/site/requests.php]:
```suggestion
					'disabled'    => $move?->isMovableTo($child) === false,
```

File: panel/src/components/Navigation/Tree.vue [hunk 3]
```diff
@@ -34,8 +35,7 @@
 					v-bind="$props"
 					:items="item.children"
 					:level="level + 1"
-					@select="select"
-					@toggle="toggle"
+					v-on="$listeners"
```
Comment (distantnative) [inline panel/src/components/Navigation/Tree.vue]: Please noooo, never `$listeners`

File: src/Cms/Page.php [hunk 4]
```diff
@@ -691,6 +692,15 @@ public function isListed(): bool
 		return $this->isPublished() && $this->num() !== null;
 	}
 
+	public function isMovableTo(Page|Site $parent): bool
+	{
+		try {
+			return PageRules::move($this, $parent);
+		} catch (Throwable $e) {
```
Comment (distantnative) [inline src/Cms/Page.php]:
```suggestion
		} catch (Throwable) {
```

File: src/Panel/Request.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+class Request
+{
+	/**
+	 * Renders request responses
+	 */
+	public static function response($data, array $options = []): Response
+	{
+		$data = Json::responseData($data);
+
+		return Panel::json(Json::responseData($data), $data['code'] ?? 200);
```
Comment (distantnative) [inline src/Panel/Request.php]: Is it intended that `$data` is passed twice through `Json::responseData()`? If so, can we add a comment to explain why?

Comment (bastianallgeier) [reply src/Panel/Request.php]: It's just sloppiness after I refactored the code.

File: panel/src/components/Navigation/Tree.vue (hunk 3 above)
Comment (bastianallgeier) [reply panel/src/components/Navigation/Tree.vue]: Sloppy again. Sorry! I just fixed it.

**Comment by distantnative** [comment]
Love it

**Pull request merged**